### PR TITLE
Fix virtualbox packer, add libvirt packer

### PR DESCRIPTION
**Pull request by bogdando**
* Use Ubuntu 14.04.3 ISO instead ovf box
* Add a libvirt target as well
* Add additional build shell scripts from
  bogdando/packer-atlas-example
* Use sudo for all shell commands
* Fix virtualbox guest additions build order
* Cleanups to reduce end image size
* Update README how to build and up both for vbox/libvirt

Signed-off-by: Bogdan Dobrelya <bdobrelia@mirantis.com>

### CODE REVIEW

**Comment by bogdando** [comment]
testing in progress

**Comment by bogdando** [comment]
packer builds for vbox & libvirt PASSED
openstack example deploy testing in progress

**Comment by bogdando** [comment]
the whole playbooks/tasks/base.yaml seem were missed in the packer ;(

**Comment by bogdando** [comment]
I cannot fix issues with the missing packages on the provisioned node. Even though the task playbooks/tasks/base.yaml was completed in the packer build log.

**Comment by bogdando** [comment]
It is strange, but that worked just fine for @loles

**Comment by pigmej** [comment]
For vbox it fails to install additions, so it's very very slow. Also it checked additions installation during `vagrant up` so I suppose this patch doesn't speedup `vagrant up` with `vbox`.

additions installation during `vagrant up` fails with:
```
Failed to install using DKMS, attempting to install without
/tmp/vbox.0/Makefile.include.header:97: *** Error: unable to find the sources of your current Linux kernel. Specify KERN_DIR=<directory> and run Make again.  Stop.
Creating user for the Guest Additions.
Creating udev rule for the Guest Additions kernel module.
``
So probably dev headers would solve this part, but it's during vagrant up, not during packer installation.

**Comment by pigmej** [comment]
After latest changes, vbox additions are working. But it seems that you need vbox 5.0.6+

File: bootstrap/playbooks/files/vagrant.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
```
Comment (loles) [inline bootstrap/playbooks/files/vagrant.sh]: Why do we need this? ssh-keys are working ok

File: bootstrap/playbooks/files/minimize.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+apt-get -y purge popularity-contest installation-report landscape-common wireless-tools wpasupplicant ubuntu-serverguide
+
+# Clean up the apt cache
+apt-get -y autoremove --purge
+apt-get -y autoclean
+apt-get -y clean
+
+# Clean up orphaned packages with deborphan
+apt-get -y install deborphan
+while [ -n "$(deborphan --guess-all --libdevel)" ]; do
```
Comment (loles) [inline bootstrap/playbooks/files/minimize.sh]: This installs too much:
```libxdmcp6
libmpdec2
python-mysqldb
python-software-properties
python-dev
python-pudb
python-mock
python-keystoneclient
software-properties-common
python-libtorrent
python-virtualenv
language-selector-common
libxau6
```

File: bootstrap/playbooks/files/minimize.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+#echo "==> Removing development tools"
+#dpkg --list | grep -i compiler | awk '{ print $2 }' | xargs apt-get -y purge
+#apt-get -y purge cpp gcc g++
+#apt-get -y purge build-essential
+#echo "==> Removing default system Ruby"
+#apt-get -y purge ruby ri doc
+#echo "==> Removing default system Python"
+#apt-get -y purge python-dbus libnl1 python-smartpm python-twisted-core libiw30 python-twisted-bin libdbus-glib-1-2 python-pexpect python-pycurl python-serial python-gobject python-pam python-openssl libffi5
+echo "==> Removing X11 libraries"
+apt-get -y purge libx11-data xauth libxmuu1 libxcb1 libx11-6 libxext6
```
Comment (loles) [inline bootstrap/playbooks/files/minimize.sh]: This removes python-pygraphviz which is required by solar

File: bootstrap/playbooks/files/minimize.sh [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+echo "==> Removing man pages"
+rm -rf /usr/share/man/*
+echo "==> Removing APT files"
+find /var/lib/apt -type f | xargs rm -f
+echo "==> Removing anything in /usr/src"
+rm -rf /usr/src/*
+echo "==> Removing any docs"
+rm -rf /usr/share/doc/*
+echo "==> Removing caches"
+find /var/cache -type f -exec rm -rf {} \;
```
Comment (loles) [inline bootstrap/playbooks/files/minimize.sh]: This removes apt cache and no package can be installed without running `apt-get update`

**Comment by bogdando** [comment]
Now the OpenStack example deploy passes the most of tasks but only the "Could not find data item neutron_plugins_ml2 in any Hiera data" error. Which seems unrelated to the PR under test. And If I run the task manually "sudo solar resource action run neutron_plugins_ml2" it works

**Comment by bogdando** [comment]
For libvirt, another task fails:
[vagrant@10.0.0.4:22] out: TASK: [shell gem install librarian-puppet --no-ri --no-rdoc] ****************** 
[vagrant@10.0.0.4:22] out: failed: [localhost] => {"changed": true, "cmd": "gem install librarian-puppet --no-ri --no-rdoc", "delta": "0:00:14.554233", "end": "2015-10-28 13:22:43.942498", "rc": 1, "start": "2015-10-28 13:22:29.388265", "warnings": []}
[vagrant@10.0.0.4:22] out: stderr: ERROR:  Error installing librarian-puppet:
[vagrant@10.0.0.4:22] out:      ERROR: Failed to build gem native extension.
[vagrant@10.0.0.4:22] out: 
[vagrant@10.0.0.4:22] out:         /usr/bin/ruby1.9.1 extconf.rb
[vagrant@10.0.0.4:22] out: /usr/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require': cannot load such file -- mkmf (LoadError)

**Comment by bogdando** [comment]
ruby-dev is missing, something is wrong with the packer scripts again

**Comment by loles** [comment]
Current version worked for me

**Comment by bogdando** [comment]
It seems lika a packer bug. I removed all mentions of the deborhpan from build scripts, but when I build from clean HEAD state, the packer still mentions deborphan in logs! Is it an invalid cached minimize.sh it keeps somewhere?

**Comment by loles** [comment]
I tested images for libvirt and for virtualbox. Both are working

**Pull request merged**